lib/ansible/modules/cloud/cloudscale/cloudscale_floating_ip.py

```diff
@@ -0,0 +1,285 @@
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.cloudscale import AnsibleCloudscaleBase
+
+
+class AnsibleCloudscaleFloatingIP(AnsibleCloudscaleBase):
+
+    def __init__(self, module, api_token):
+        super(AnsibleCloudscaleFloatingIP, self).__init__(module, api_token)
+
+        # Initialize info dict
+        # Set state to absent, will be updated by self.update_info()
+        self.info = {'state': 'absent'}
+
+        if self._module.params['ip']:
+            self.update_info()
+
+    @staticmethod
+    def _resp2info(resp):
+        # If the API response has some content, the floating IP must exist
+        resp['state'] = 'present'
+
+        # Add the IP address to the response, otherwise handling get's to complicated as this
+        # has to be converted from the network all the time.
+        resp['ip'] = str(ip_network(resp['network']).network_address)
+
+        # Replace the server with just the UUID, the href to the server is useless and just makes
+        # things more complicated
+        resp['server'] = resp['server']['uuid']
+
+        return resp
+
+    def update_info(self):
+        resp = self._get('floating-ips/' + self._module.params['ip'])
+        if resp:
+            self.info = self._resp2info(resp)
+        else:
+            self.info = {'ip': self._module.params['ip'],
+                         'state': 'absent'}
+
+    def request_floating_ip(self):
+        params = self._module.params
+
+        # check for required parameters to request a floating IP
+        missing_parameters = []
+        for p in ('ip_version', 'server'):
+            if p not in params or not params[p]:
+                missing_parameters.append(p)
+
+        if len(missing_parameters) > 0:
+            self._module.fail_json(msg='Missing required parameter(s) to request a floating IP: %s.' %
+                                   ' '.join(missing_parameters))
+
+        data = {'ip_version': params['ip_version'],
+                'server': params['server']}
+
+        if params['prefix_length']:
+            data['prefix_length'] = params['prefix_length']
+        if params['reverse_ptr']:
+            data['reverse_ptr'] = params['reverse_ptr']
+
+        self.info = self._resp2info(self._post('floating-ips', data))
+
+    def release_floating_ip(self):
+        self._delete('floating-ips/%s' % self._module.params['ip'])
+        self.info = {'ip': self.info['ip'], 'state': 'absent'}
+
+    def update_floating_ip(self):
+        params = self._module.params
+        if 'server' not in params or not params['server']:
+            self._module.fail_json(msg='Missing required parameter to update a floating IP: server.')
+        self.info = self._resp2info(self._post('floating-ips/%s' % params['ip'], {'server': params['server']}))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=('present', 'absent')),
+            ip=dict(),
```

Comment:
according to doc spec, there should be an alias:

~~~diff
-            ip=dict(),
+            ip=dict(aliases=['network']),
~~~